l10n_ch_payment_register/wizard/payment_order_create.py

```diff
@@ -0,0 +1,144 @@
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import time
+from lxml import etree
+
+from openerp import models, fields, api, _
+
+
+class PaymentOrderCreate(models.TransientModel):
+    """
+    Create a payment object with lines corresponding to the account move line
+    to pay according to the date and the mode provided by the user.
+    Hypothesis:
+    - Small number of non-reconciled move line, payment mode and
+    bank account type,
+    - Big number of partner and bank account.
+
+    If a type is given, unsuitable account Entry lines are ignored.
+    """
+
+    _name = 'payment.order.create'
+    _description = 'payment.order.create'
+
+    duedate = fields.Date('Due Date', required=True,
+                          default=lambda *a: time.strftime('%Y-%m-%d'))
+
+    entries = fields.Many2many('account.move.line', 'line_pay_rel', 'pay_id',
+                               'line_id')
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        res = super(PaymentOrderCreate, self).\
+            fields_view_get(view_id=view_id, view_type=view_type,
+                            toolbar=toolbar, submenu=False)
+
+        context = dict(self._context or {})
+        if context and 'line_ids' in context:
+            doc = etree.XML(res['arch'])
+            nodes = doc.xpath("//field[@name='entries']")
+            for node in nodes:
+                node.set('domain', '[("id", "in", ' +
+                         str(context['line_ids']) + ')]')
+            res['arch'] = etree.tostring(doc)
+        return res
+
+    @api.one
+    def create_payment(self):
+        order_obj = self.env['payment.register']
+        line_obj = self.env['account.move.line']
+        reg_line_obj = self.env['payment.register.line']
+
+        context = dict(self._context or {})
+        line_ids = [entry.id for entry in self.entries]
+        if not line_ids:
+            return {'type': 'ir.actions.act_window_close'}
+
+        payment = order_obj.browse(context['active_id'])
+        line2bank = line_obj.line2bank(line_ids)
+
+        # # Finally populate the current payment with new lines:
+        for line in line_obj.browse(line_ids):
+            if payment.date_prefered == "now":
+                # no payment date => immediate payment
+                date_to_pay = False
+            elif payment.date_prefered == 'due':
+                date_to_pay = line.date_maturity
+            elif payment.date_prefered == 'fixed':
+                date_to_pay = payment.date_scheduled
+            reg_line_obj.\
```

Comment:
A prefered coding style would be

    reg_line_obj.create(
        {'name': ...,
         'other_key': ...})